vercel/resource_project.go

```diff
@@ -176,21 +175,14 @@ At this time you cannot use a Vercel Project resource with in-line ` + "`environ
 			"vercel_authentication": schema.SingleNestedAttribute{
 				Description: "Ensures visitors to your Preview Deployments are logged into Vercel and have a minimum of Viewer access on your team.",
 				Optional:    true,
-				Computed:    true,
-				Default: objectdefault.StaticValue(types.ObjectValueMust(
-					map[string]attr.Type{
-						"protect_production": types.BoolType,
-					},
-					map[string]attr.Value{
-						"protect_production": types.BoolValue(false),
-					},
-				)),
 				Attributes: map[string]schema.Attribute{
-					"protect_production": schema.BoolAttribute{
-						Description: "If true, production deployments will also be protected",
-						Optional:    true,
-						Computed:    true,
-						Default:     booldefault.StaticBool(false),
+					"deployment_type": schema.StringAttribute{
+						Required:      true,
+						Description:   "The deployment environment to protect. Must be one one `standard_protection`, `all_deployments`, or `only_preview_deployments`.",
+						PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+						Validators: []validator.String{
+							stringOneOf("standard_protection", "all_deployments", "only_preview_deployments"),
+						},
```

Comment:
I don't think this needs to be netsted under `vercel_authentication` any more? We could just have 
```hcl
resource "vercel_project" "example" {
    name = "some-project"
    vercel_authentication = "all_deployments"
}
```
WDYT?

Reply:
we can then just have a `Default` on it? :)

Reply:
I think we should make the terraform reflect the api structure closeishly, otherwise if vercel auth gets a new settings we'll need another breaking change